src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -41,6 +44,21 @@ public void register() {
     }
 
     /**
+     * Remove everything in the pipeline
+     */
+    public void clear() {
+        this.steps.stream()
+                .map(StepRemovedEvent::new)
+                .collect(Collectors.toList())
```

Comment:
Sending a `StepRemovedEvent` causes `this.steps` to be modified, but bad things happen when streams modify their source.   So, it has to be a `forEach` on a separate list, not on a stream of that list.